### PR TITLE
Some code refinements in MOSART-heat

### PR DESCRIPTION
**Pull request by liho745**
The latent heat formula for water vaporization from Wu et al. (2012) has been replaced by that 
from Dingman (2009). Also adjusted a couple of parameters to be more reasonable.  

[BFB]

### CODE REVIEW

**Comment by rljacob** [comment]
is this for v2 BGC?

**Comment by liho745** [comment]
So far, based on what I know, it is not linked to any other V2 BGC components, and not included in any BGC experiments.

**Comment by bishtgautam** [comment]
@liho745, I have reviewed the PR and made few comments for you to address.
@hydrotian, Could you also review this PR? thanks.

File: components/mosart/src/riverroute/MOSART_physics_mod.F90 [hunk 1]
```diff
@@ -448,6 +451,13 @@ subroutine Euler
           if (ctlSubwWRM%ExtractionFlag > 0 ) then
              call t_startf('mosartr_wrm_ERFlow')
              call ExtractionRegulatedFlow(localDeltaT)
+             ! a simple treatment after extracting water from the regulated streamflow. Assuming the extraction won't change the water temperature in the release
+             ! but the heat flux will be changed due to chaning streamflow
```
Comment (hydrotian) [inline components/mosart/src/riverroute/MOSART_physics_mod.F90:455]: chaning -> changing

**Comment by rljacob** [comment]
Can you change the title of this PR so its not the branch name?

**Comment by liho745** [comment]
@bishtgautam , @hydrotian , @rljacob : I've addressed all your comments, and committed my new changes.

**Pull request merged**